packages/shared/src/getToken.ts

```diff
@@ -0,0 +1,160 @@
+import { inBrowser } from './browser';
+import type { ClerkStatus, GetTokenOptions, LoadedClerk } from './types';
+
+const POLL_INTERVAL_MS = 50;
+const MAX_POLL_RETRIES = 100; // 5 seconds of polling
+const TIMEOUT_MS = 10000; // 10 second absolute timeout
```

Comment:
Consider using the existing `retry()` helper: https://github.com/clerk/javascript/blob/main/packages/shared/src/retry.ts